src/dune_engine/scheduler.ml

```diff
@@ -759,45 +782,62 @@ let maybe_clear_screen ~config =
          ])
 
 let poll ?config ~once ~finally () =
-  let t = prepare ?config () in
-  let watcher = File_watcher.create () in
-  let block_waiting_for_changes () =
-    match Event.next () with
-    | Job_completed _ -> assert false
-    | Files_changed -> Continue
-    | Signal signal ->
-      got_signal signal;
-      Exit
-  in
-  let wait msg =
-    Console.Status_line.set_temporarily
-      (fun () ->
-        Some (Pp.seq msg (Pp.verbatim ", waiting for filesystem changes...")))
-      (fun () ->
-        let res = block_waiting_for_changes () in
-        maybe_clear_screen ~config;
-        res)
+  let t = prepare ?config ~polling:true () in
+  let finally () =
+    (* Re-create the throttle so that we don't keep old ivars from one run to
+       the next *)
+    t.job_throttle <- Fiber.Throttle.create (Fiber.Throttle.size t.job_throttle);
```

Comment:
Why do we need to re-create the throttle? I thought the idea was that we don't leave the ivars hanging, so the existing throttle should work fine?

Reply:
That's true. Will get rid of this.